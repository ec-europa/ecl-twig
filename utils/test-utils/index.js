const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
