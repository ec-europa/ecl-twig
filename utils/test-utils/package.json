{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
