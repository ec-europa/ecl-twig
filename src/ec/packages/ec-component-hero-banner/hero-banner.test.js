import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

import {
  dataDefault,
  dataImage,
  dataImageShade,
  dataPrimary,
  dataLeft,
} from './demo/data';

function formatBanner(b) {
  const iconType = b.link.icon.shape.split('--');
  const banner = {
    type: b.variant,
    title: b.title,
    description: b.description,
    link: {
      link: {
        label: b.link.label,
        icon_position: 'after',
      },
      icon: {
        type: iconType[0],
        name: iconType[1],
        transform: b.link.icon.transform,
        size: b.link.icon.size,
        path: 'static/icons.svg',
      },
    },
    centered: b.centered,
  };

  if ('image' in b) {
    banner.image = 'https://v2--europa-component-library.netlify.com'.concat(
      b.image
    );
  }

  return banner;
}

describe('EC - Hero Banner', () => {
  const template =
    '@ecl-twig/ec-component-hero-banner/ecl-hero-banner.html.twig';
  const render = params => renderTwigFileAsNode(template, params);

  describe('default', () => {
    const data = formatBanner(dataDefault);

    test(`- renders correctly`, () => {
      expect.assertions(1);

      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('image', () => {
    const data = formatBanner(dataImage);

    test(`- renders correctly`, () => {
      expect.assertions(1);

      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('image-shade', () => {
    const data = formatBanner(dataImageShade);

    test(`- renders correctly`, () => {
      expect.assertions(1);

      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('primary', () => {
    const data = formatBanner(dataPrimary);

    test(`- renders correctly`, () => {
      expect.assertions(1);

      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('align-left', () => {
    const data = formatBanner(dataLeft);

    test(`- renders correctly`, () => {
      expect.assertions(1);

      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });
});
