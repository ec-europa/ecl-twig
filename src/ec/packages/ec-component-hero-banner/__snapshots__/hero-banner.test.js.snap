// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Hero Banner align-left - renders correctly 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--default"
>
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (default)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner align-left renders correctly with extra attributes 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--default"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (default)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner align-left renders correctly with extra class names 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--default custom-class custom-class--test"
>
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (default)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner default - renders correctly 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--default ecl-hero-banner--centered"
>
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (default)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner default renders correctly with extra attributes 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--default ecl-hero-banner--centered"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (default)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner default renders correctly with extra class names 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--default ecl-hero-banner--centered custom-class custom-class--test"
>
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (default)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner image - renders correctly 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--image ecl-hero-banner--centered"
>
  <div
    class="ecl-hero-banner__image"
    style="background-image:url(https://v2--europa-component-library.netlify.com/example-image.jpg)"
  />
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (image)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner image renders correctly with extra attributes 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--image ecl-hero-banner--centered"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-hero-banner__image"
    style="background-image:url(https://v2--europa-component-library.netlify.com/example-image.jpg)"
  />
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (image)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner image renders correctly with extra class names 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--image ecl-hero-banner--centered custom-class custom-class--test"
>
  <div
    class="ecl-hero-banner__image"
    style="background-image:url(https://v2--europa-component-library.netlify.com/example-image.jpg)"
  />
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (image)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner image-shade - renders correctly 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--image-shade ecl-hero-banner--centered"
>
  <div
    class="ecl-hero-banner__image"
    style="background-image:url(https://v2--europa-component-library.netlify.com/example-image.jpg)"
  />
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (image shade)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner image-shade renders correctly with extra attributes 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--image-shade ecl-hero-banner--centered"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-hero-banner__image"
    style="background-image:url(https://v2--europa-component-library.netlify.com/example-image.jpg)"
  />
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (image shade)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner image-shade renders correctly with extra class names 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--image-shade ecl-hero-banner--centered custom-class custom-class--test"
>
  <div
    class="ecl-hero-banner__image"
    style="background-image:url(https://v2--europa-component-library.netlify.com/example-image.jpg)"
  />
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (image shade)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner primary - renders correctly 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--primary ecl-hero-banner--centered"
>
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (primary)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner primary renders correctly with extra attributes 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--primary ecl-hero-banner--centered"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (primary)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`EC - Hero Banner primary renders correctly with extra class names 1`] = `
<section
  class="ecl-hero-banner ecl-hero-banner--primary ecl-hero-banner--centered custom-class custom-class--test"
>
  <div
    class="ecl-container ecl-hero-banner__container"
  >
    <div
      class="ecl-hero-banner__content"
    >
      <h1
        class="ecl-hero-banner__title"
      >
        EU Budget for the future (primary)
      </h1>
      <p
        class="ecl-hero-banner__description"
      >
        The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
      </p>
      <button
        class="ecl-button ecl-button--call ecl-hero-banner__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Subscribe
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</section>
`;
